### DISCUSSION TITLE
Python Minor Version Deprecation Policy

### DISCUSSION THREAD

**Discussion by StijnKas** [pegasystems/pega-datascientist-tools#199] [General]
Here's an open question: according to the Numpy Python depcrecation policy, support for 3.9 will be dropped next month: https://numpy.org/neps/nep-0029-deprecation_policy.html. Currently, we support 3.8 (which is not even supported by numpy at this point). That policy doesn't guarantee it _won't_ work on older versions, it just guarantees that versions _above_ the minimum will be supported.

Do we want to tie our minimum supported version to the Numpy policy as well? Given that we have numpy as a dependency, I think it makes sense to hold onto their guidelines as the 'smallest' denominator. If they drop support and something breaks, there's nothing we can do to fix that. Thoughts @operdeck  /  @yusufuyanik1 ?

**Comment by operdeck** [comment]
On the "something breaks" - wouldn't we be covered through unit tests?

**Comment by StijnKas** [comment]
This now becomes more concrete: Numpy recently updated to 2.0, which supports 3.9 and up. There are backwards incompatible changes, so to 'support' the latest version of numpy, you need to tie to >=2.0. Our dependencies like Polars will surely do this soon, meaning by transitive requirements, we will only be compatible with versions <= 3.9. Just an fyi really that this will need to happen soon.
